File: Gemini/1. Algoritmo Básico Ordenamiento por Burbuja/Javascript/burbuja.js
function bubbleSort(arr) {
  const n = arr.length;
  for (let i = 0; i < n - 1; i++) {
    for (let j = 0; j < n - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
      }
    }
  }
  return arr;
}

const entrada = prompt("Ingrese una lista de números enteros separados por espacios:");
const numeros = entrada.split(" ").map(Number);
const listaOrdenada = bubbleSort(numeros);
console.log("Lista ordenada:", listaOrdenada);

File: Gemini/1. Algoritmo Básico Ordenamiento por Burbuja/Ruby/burbuja.rb
def bubble_sort(array)
  n = array.length
  loop do
    swapped = false
    (n - 1).times do |i|
      if array[i] > array[i + 1]
        array[i], array[i + 1] = array[i + 1], array[i]
        swapped = true
      end
    end
    break unless swapped
  end
  array
end

print "Ingrese una lista de números enteros separados por espacios: "
entrada = gets.chomp
numeros = entrada.split(" ").map(&:to_i)
lista_ordenada = bubble_sort(numeros)
puts "Lista ordenada: #{lista_ordenada}"